src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static updateAliveAnimals() {
+    this.alive = this.alive.filter(animal => animal.health > 0);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
+};
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(someone) {
+    if (someone instanceof Herbivore && !someone.hidden) {
+      someone.health -= 50;
+    }
+    Animal.updateAliveAnimals();
```

Comment:
You're updating the alive animals array every time a bite method is called. Consider updating the array when the health of an animal changes to 0 or less, instead of after every bite.